File: src/main/java/com/sce/data/gaia/dao/domain/CustomUser.java
package com.sce.data.gaia.dao.domain;


import lombok.Data;

import javax.persistence.*;

/**
 * system user table
 * @author bk201
 */
@Entity
@Table(name = "bi_gaia_users")
@Data
public class CustomUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @Column(table = "bi_gaia_users",name = "user_name",nullable = false,unique = true)
    private String userName;
    @Column(table = "bi_gaia_users",name = "password",nullable = false)
    private String password;
    @Column(table = "bi_gaia_users",name = "role_names",nullable = false)
    private String roleNames;
}

File: src/main/java/com/sce/data/gaia/dao/domain/CustomUser.kt
package com.sce.data.gaia.dao.domain


import com.sce.data.gaia.constant.ColumnNames
import com.sce.data.gaia.constant.TableNames

import javax.persistence.*

/**
 * system user table
 * @author bk201
 */
@Entity
@Table(name = TableNames.CUSTOM_SYS_USERS)
class CustomUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0
    @Column(table = TableNames.CUSTOM_SYS_USERS, name = ColumnNames.USER_NAME, nullable = false, unique = true)
    var userName: String? = null
    @Column(table = TableNames.CUSTOM_SYS_USERS, name = ColumnNames.PASSWORD, nullable = false)
    var password: String? = null
    @Column(table = TableNames.CUSTOM_SYS_USERS, name = ColumnNames.ROLE_NAMES, nullable = false)
    var roleNames: String? = null
}